### PR TITLE
Add support for Plone 5.1.x.

### PR DESCRIPTION
**Pull request by mbaechtold**
This also removes support for Plone 4.2.x (which is no longer maintained).

### CODE REVIEW

File: test-plone-5.1.x.cfg [hunk 1]
```diff
@@ -1,9 +1,15 @@
...
-[versions]
-ftw.builder = 1.4.0
+extensions =
+    mr.developer
+
+auto-checkout =
+    ftw.testbrowser
+
+[branches]
+ftw.testbrowser = plone-5.1.x
```
Comment (mbaechtold) [inline test-plone-5.1.x.cfg]: 🚧 `ftw.testbrowser` has not Plone 5.1.x release yet, see https://github.com/4teamwork/ftw.testbrowser/pull/47

File: ftw/footer/tests/test_footer_functional.py [hunk 2]
```diff
@@ -19,16 +18,35 @@ def setUp(self):
...
+
+        # The footer portlet is shown on the Plone Site.
+        browser.open(self.portal)
+        self.assertEqual(
+            'My Portlet Text',
+            browser.css('#footer-column-1').first.text
+        )
+
+        # The footer portlet is shown on the control panel.
+        browser.open(self.portal.absolute_url() + '/@@overview-controlpanel')
```
Comment (maethu) [inline ftw/footer/tests/test_footer_functional.py]: This is no longer the same test. The control panel is on a contentish object (Plone Root)
But it should also work, for example on a portlet context: `http://nohost/plone/++contextportlets++ftw.footer.column1/portlet_static/edit`

Comment (maethu) [inline ftw/footer/tests/test_footer_functional.py]: This is no longer the same test. The control panel is on a contentish object (Plone Root)
But it should also work, for example on a portlet context: `http://nohost/plone/++contextportlets++ftw.footer.column1/portlet_static/edit`

Comment (mbaechtold) [reply ftw/footer/tests/test_footer_functional.py]: @maethu, I've rewritten the test class, please review it in its entirety.

Comment (maethu) [reply ftw/footer/tests/test_footer_functional.py]: ahhhhhhhhhh sry... 🙈

**Comment by maethu** [comment]
@mbaechtold You can merge this one, including ftw.testing from source (plone 5.1 branch)

File: ftw/footer/tests/test_footer_functional.py [hunk 3]
```diff
@@ -17,18 +16,61 @@ def setUp(self):
...
+        # The footer portlet is shown on the control panel, which is a view
+        # on a contentish object (Plone Site).
+        browser.open(self.portal.absolute_url() + '/@@overview-controlpanel')
+        self.assertEqual(
+            'My Portlet Text',
+            browser.css('#footer-column-1').first.text
+        )
+
+        # The footer portlet is shown when managing the footer.
+        browser.find('Manage Footer').click()
```
Comment (mbaechtold) [inline ftw/footer/tests/test_footer_functional.py]: @maethu, should this go into `self.test_footer_on_non_contentish_object()`?

**Pull request merged**